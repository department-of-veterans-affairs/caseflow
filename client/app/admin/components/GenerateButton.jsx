/* eslint-disable func-style */
import React, { useEffect, useState } from 'react';
import Button from '../../components/Button';
import Modal from '../../components/Modal';
import Alert from '../../components/Alert';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from 'app/constants/AppConstants';
import PropTypes from 'prop-types';
import { CSVLink } from 'react-csv';

const GenerateButton = (props) => {

  // state properties
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [showEmptyResultsBanner, setShowEmptyResultsBanner] = useState(false);
  const [showErrorBanner, setShowErrorBanner] = useState(false);
  const [extractedResults, setExtractedResults] = useState('');

  useEffect(() => {
    setIsLoading(props.isLoading);
  }, [props.isLoading]);

  useEffect(() => {
    setExtractedResults(props.extractedResults);

    if (props.extractedResults?.length > 0) {
      setShowEmptyResultsBanner(false);
      setShowConfirmModal(true);
    }
  }, [props.extractedResults]);

  useEffect(() => {
    if (props.emptyResultsMessage?.length > 0) {
      setShowEmptyResultsBanner(true);
      setShowConfirmModal(false);
    }
  }, [props.emptyResultsMessage]);

  useEffect(() => {
    if (props.manualExtractionSuccess === false) {
      setShowErrorBanner(true);
    } else {
      setShowErrorBanner(false);
    }
  }, [props.manualExtractionSuccess]);

  const onClickGenerate = () => {
    props.sendExtractRequest();
  };

  return (
    <div style={{ height: '75vh' }}>
      {
        showEmptyResultsBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="No Veterans were found" type="success" />
        </div>
      }
      {
        showErrorBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="Veteran Extract Failed" type="error" />
        </div>
      }
      {
        !isLoading &&
        <Button
          id="generate-extract"
          onClick={() => onClickGenerate()}
        >
          Generate Veteran Extract
        </Button>
      }
      {
        showConfirmModal &&

        <Modal title="This file contains PII"
          confirmButton={<CSVLink data={extractedResults} filename="veteran_extract.csv" onClick={() => setShowConfirmModal(false)}>Confirm</CSVLink>}
          closeHandler={() => {
            setShowConfirmModal(false);
          }}
        >
          Are you sure you want to download it?
        </Modal>
      }
      {
        isLoading &&

        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Action is Running...
          </div>
        </LoadingContainer>
      }
    </div>
  );
};

GenerateButton.propTypes = {
  extractedResults: PropTypes.string,
  manualExtractionSuccess: PropTypes.bool,
  isLoading: PropTypes.bool,
  sendExtractRequest: PropTypes.func,
  emptyResultsMessage: PropTypes.string,
};

export default GenerateButton;
