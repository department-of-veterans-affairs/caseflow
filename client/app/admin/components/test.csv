VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN