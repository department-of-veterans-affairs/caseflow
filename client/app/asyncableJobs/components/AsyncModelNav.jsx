import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import DropdownButton from '../../components/DropdownButton';

const DATE_TIME_FORMAT = 'ddd MMM DD HH:mm:ss YYYY';

export default class AsyncModelNav extends React.PureComponent {
  modelNameLinks = () => {
    const links = [];
    const modelNames = this.props.models.sort();
    const numLinks = modelNames.length;

    for (let modelName of modelNames) {
      const url = `/asyncable_jobs/${modelName}/jobs`;
      let modelLink;

      if (numLinks > 4) {
        modelLink = {
          title: modelName,
          target: url
        };
      } else {
        modelLink = <span key={modelName} className="cf-model-jobs-link"><a href={url}>{modelName}</a></span>;
      }

      links.push(modelLink);
    }

    return numLinks > 4 ? <DropdownButton lists={links} label="Filter by Job Type" /> : links;
  }

  render = () => {
    const currentPath = this.props.asyncableJobKlass ?
      `/asyncable_jobs/${this.props.asyncableJobKlass}/jobs` :
      '/jobs';

    return <div>
      <strong>Last updated:</strong> {moment(this.props.fetchedAt).format(DATE_TIME_FORMAT)}
      <div style={{ marginTop: '.5em' }}>
        <a style={{ marginRight: '.5em' }} href="/jobs" className="cf-link-btn">All jobs</a>
        {this.modelNameLinks()}
        <a style={{ float: 'right' }} href={`${currentPath}.csv`} className="cf-link-btn">Download as CSV</a>
      </div>
    </div>;
  }
}

AsyncModelNav.propTypes = {
  models: PropTypes.array,
  fetchedAt: PropTypes.string,
  asyncableJobKlass: PropTypes.string
};
