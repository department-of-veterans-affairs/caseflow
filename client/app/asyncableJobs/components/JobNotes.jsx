import React from 'react';
import Button from '../../components/Button';
import ApiUtil from '../../util/ApiUtil';
import PropTypes from 'prop-types';
import moment from 'moment';
import ReactMarkdown from 'react-markdown';

const DATE_TIME_FORMAT = 'ddd MMM DD HH:mm:ss YYYY';

import Checkbox from '../../components/Checkbox';
import TextareaField from '../../components/TextareaField';

class NewNoteForm extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      saveInProgress: false,
      checkboxSelected: true,
      newNote: ''
    };
  }

  onSave = () => {
    const job = this.props.job;
    const url = `/asyncable_jobs/${job.klass}/jobs/${job.id}/note`;
    const form = this;
    const notes = this.props.notes;

    const data = {
      note: this.state.newNote,
      send_to_intake_user: this.state.checkboxSelected
    };

    notes.setState({ reloaded: false });

    ApiUtil.post(url, { data }).
      then(
        (response) => {
          const newNote = JSON.parse(response.text);

          form.setState({
            saveInProgress: false,
            checkboxSelected: false,
            newNote: ''
          });

          // add note to current page.
          notes.props.notes.unshift(newNote);
          notes.setState({ reloaded: true });
        }
      );
  }

  onChange = (event) => {
    this.setState({ newNote: event });
  }

  onCheckboxChange = (event) => {
    this.setState({ checkboxSelected: event });
  }

  isSaveDisabled = () => {
    return this.state.saveInProgress || this.state.newNote.trim().length === 0;
  }

  isCheckboxChecked = () => {
    return this.state.checkboxSelected;
  }

  sendToUserCheckbox = () => {
    return <Checkbox
      label="Send as message to user"
      name="send_to_intake_user"
      value={this.isCheckboxChecked()}
      onChange={this.onCheckboxChange}
      disabled={this.state.saveInProgress}
    />;
  }

  render = () => {
    return <div className="comment-size-container">
      <TextareaField
        hideLabel
        name="Add Note"
        aria-label="Add Note"
        onChange={this.onChange}
        value={this.state.newNote}
      />
      <div className="comment-save-button-container">
        <span className="cf-right-side">
          { this.props.job.user && this.sendToUserCheckbox() }
          <Button
            name="save"
            disabled={this.isSaveDisabled()}
            onClick={this.onSave}>
              Add Note
          </Button>
        </span>
      </div>
    </div>;
  }

}

NewNoteForm.propTypes = {
  job: PropTypes.object,
  notes: PropTypes.object,
  onSave: PropTypes.func,
  onCancel: PropTypes.func
};

export default class JobNotes extends React.PureComponent {

  render = () => {
    const { notes, job } = this.props;

    return <div className="job-notes">
      <h3>Notes</h3>
      <NewNoteForm job={job} notes={this} />
      <div>
        { notes.map((note, index) => {
          return <div className="job-note" key={`job-note-container-${index}`}>
            <div
              className="job-note-details"
              data-key={`job-note-${index}`}
              key={`job-note-${index}`}
              id={`job-note-${note.id}`}>

              <div>
                <span className="job-note-time">{moment(note.created_at).format(DATE_TIME_FORMAT)}</span>
                <span className="job-note-user">{note.user}</span>
              </div>
              <div className="job-note-note"><ReactMarkdown source={note.note} /></div>

            </div>
          </div>;
        })}
      </div>
    </div>;
  }
}

JobNotes.propTypes = {
  job: PropTypes.object,
  notes: PropTypes.array
};
