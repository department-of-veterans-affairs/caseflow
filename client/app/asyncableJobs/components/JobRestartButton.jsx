import React from 'react';
import Button from '../../components/Button';
import ApiUtil from '../../util/ApiUtil';
import PropTypes from 'prop-types';

class JobRestartButton extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      restarted: false,
      restarting: false
    };
  }

  restartJob = () => {
    this.setState({ restarting: true });
    this.sendRestart();
  }

  sendRestart = () => {
    const button = this;
    const page = this.props.page;
    const job = this.props.job;
    const url = `/asyncable_jobs/${job.klass}/jobs/${job.id}`;

    ApiUtil.patch(url, {}).
      then(
        (response) => {
          const responseObject = JSON.parse(response.text);

          Object.assign(job, responseObject);

          job.error = '';
          job.restarted = true;

          button.setState({
            restarted: true,
            restarting: false
          });

          // trigger table rerender
          const jobsRestarted = page.state.restarted + 1;

          page.setState({ restarted: jobsRestarted });
        }
      );
  }

  getButtonClassNames = () => {
    let classNames = ['usa-button'];

    if (this.state.restarting || this.state.restarted) {
      classNames.push('usa-button-disabled');
    }

    return classNames;
  }

  getButtonText = () => {
    let txt = 'Restart';

    if (this.state.restarting) {
      txt = 'Restarting';
    } else if (this.state.restarted) {
      txt = 'Restarted';
    } else if (this.props.job.processed_at) {
      txt = 'Processed';
    } else if (this.disableRestart()) {
      txt = 'Queued';
    }

    return txt;
  }

  disableRestart = () => {
    const job = this.props.job;

    if (job.processed_at) {
      return true;
    }

    if (!job.attempted_at) {
      return true;
    }

    const fiveMinutes = 300000;

    const lastAttempted = new Date(job.attempted_at).getTime();
    const submittedAt = new Date(job.last_submitted_at).getTime();
    const now = new Date().getTime();

    if ((now - lastAttempted) < fiveMinutes || (now - submittedAt) < fiveMinutes) {
      return true;
    }

    return false;
  }

  render = () => {
    const job = this.props.job;

    return <Button
      id={`job-${job.klass}-${job.id}`}
      title={`${job.klass} ${job.id}`}
      loading={this.state.restarting}
      loadingText="Restarting..."
      disabled={this.disableRestart()}
      onClick={() => {
        this.restartJob();
      }}
      classNames={this.getButtonClassNames()}
    >{this.getButtonText()}</Button>;
  }
}

JobRestartButton.propTypes = {
  job: PropTypes.object,
  page: PropTypes.object
};

export default JobRestartButton;
