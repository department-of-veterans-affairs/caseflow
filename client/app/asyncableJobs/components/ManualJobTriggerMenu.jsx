import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../../components/Button';
import StringUtil from '../../util/StringUtil';
import classNames from 'classnames';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from '../../constants/AppConstants';

const ManualJobTriggerMenu = (props) => {

  const [loading, setLoading] = useState(false);
  const [manualJobType, setManualJobType] = useState(null);

  useEffect(() => {
    setManualJobType(props.manualJobType);
    if (props.manualJobType) {
      setLoading(false);
    }

  }, [props.manualJobType]);

  const renderMessage = () => {
    if (manualJobType) {
      const manualJobBannerClasses = classNames({
        success: props.manualJobSuccess,
        fail: !props.manualJobSuccess,
      }, 'manual-job-banner');

      const status = props.manualJobSuccess ? 'succeeded' : 'failed';

      return (
        <div className={manualJobBannerClasses}>
          Manual run of <strong>{StringUtil.snakeCaseToCapitalized(manualJobType)}</strong> has{' '}
          <strong>{status}</strong>
        </div>
      );
    }

    return null;
  };

  return (
    <div className="tab-border scheduled-jobs-tab">
      <br />
      <h1>Manually Perform Scheduled Async Jobs</h1>
      {loading &&
        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Job is Running...
          </div>
        </LoadingContainer>
      }
      {!loading && (<>
        {renderMessage()}
        <table>
          <tbody className="manual-job-table">
            {props.supportedJobs.map((jobType) => (
              <tr key={`${jobType}-row`}>
                <td>
                  <h3>{StringUtil.snakeCaseToCapitalized(jobType)}</h3>
                </td>
                <td>
                  <Button
                    name={`trigger-${jobType}-job`}
                    onClick={() => {
                      setLoading(true);
                      props.sendJobRequest(jobType);
                    }}
                  >
                Perform Now
                  </Button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </>)}
    </div>
  );
};

ManualJobTriggerMenu.propTypes = {
  supportedJobs: PropTypes.arrayOf(
    PropTypes.string
  ),
  sendJobRequest: PropTypes.func.isRequired,
  manualJobStatus: PropTypes.number,
  manualJobSuccess: PropTypes.bool,
  manualJobType: PropTypes.string,
};

export default ManualJobTriggerMenu;
