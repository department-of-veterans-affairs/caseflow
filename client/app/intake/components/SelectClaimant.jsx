import React from 'react';
import RadioField from '../../components/RadioField';
import SearchableDropdown from '../../components/SearchableDropdown';
import { BOOLEAN_RADIO_OPTIONS,
  BOOLEAN_RADIO_OPTIONS_DISABLED_FALSE,
  DECEASED_PAYEE_CODES,
  LIVING_PAYEE_CODES } from '../constants';
import COPY from '../../../COPY.json';

export default class SelectClaimant extends React.PureComponent {
  handlePayeeCodeChange(event) {
    this.props.setPayeeCode(event ? event.value : null);
  }

  shouldShowPayeeCode = () => {
    const { formType, benefitType } = this.props;

    return formType !== 'appeal' &&
      (benefitType === 'compensation' || benefitType === 'pension');
  }

  render = () => {
    const {
      isVeteranDeceased,
      veteranIsNotClaimant,
      veteranIsNotClaimantError,
      setVeteranIsNotClaimant,
      claimant,
      claimantError,
      setClaimant,
      relationships,
      payeeCode,
      payeeCodeError
    } = this.props;

    const hasRelationships = relationships.length > 0;
    let showClaimants = ['true', true].includes(veteranIsNotClaimant);

    const email = React.createElement(
      'a', { href: 'mailto:VACaseflowOps@va.gov?Subject=Add%20claimant%20to%20Corporate%20Database' }, 'email'
    );
    const claimantLabel = React.createElement(
      'p', { id: 'claimantLabel' }, COPY.CLAIMANT_NOT_FOUND_START, email, COPY.CLAIMANT_NOT_FOUND_END
    );
    const noClaimantsCopy = React.createElement(
      'p', { id: 'noClaimants',
        className: 'cf-red-text' }, COPY.NO_RELATIONSHIPS, email, COPY.CLAIMANT_NOT_FOUND_END
    );

    const claimantOptions = () => {
      return <div className="cf-claimant-options">
        <RadioField
          name="claimant-options"
          label={claimantLabel}
          strongLabel
          vertical
          options={relationships}
          onChange={setClaimant}
          value={claimant}
          errorMessage={claimantError}
        />

        {
          this.shouldShowPayeeCode() && <SearchableDropdown
            name="cf-payee-code"
            strongLabel
            label="What is the payee code for this claimant?"
            placeholder="Select"
            options={isVeteranDeceased ? DECEASED_PAYEE_CODES : LIVING_PAYEE_CODES}
            value={payeeCode}
            errorMessage={payeeCodeError}
            onChange={(event) => this.handlePayeeCodeChange(event)} />
        }

      </div>;
    };

    let veteranClaimantOptions = BOOLEAN_RADIO_OPTIONS;

    if (isVeteranDeceased) {
      // disable veteran claimant option if veteran is deceased
      veteranClaimantOptions = BOOLEAN_RADIO_OPTIONS_DISABLED_FALSE;
      // set claimant value to someone other than the veteran
      setVeteranIsNotClaimant('true');
    }

    return <div className="cf-different-claimant">
      <RadioField
        name="different-claimant-option"
        label="Is the claimant someone other than the Veteran?"
        strongLabel
        vertical
        options={veteranClaimantOptions}
        onChange={setVeteranIsNotClaimant}
        errorMessage={veteranIsNotClaimantError}
        value={veteranIsNotClaimant === null ? null : veteranIsNotClaimant.toString()}
      />

      { showClaimants && hasRelationships && claimantOptions() }
      { showClaimants && !hasRelationships && noClaimantsCopy }

    </div>;
  }
}
