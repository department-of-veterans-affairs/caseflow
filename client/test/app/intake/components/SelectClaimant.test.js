import React from 'react';
import { screen, render } from '@testing-library/react';
import { axe } from 'jest-axe';

import COPY from 'app/../COPY';

import { SelectClaimant } from 'app/intake/components/SelectClaimant';

const defaultRelationships = [
  {
    value: 'CLAIMANT_WITH_PVA_AS_VSO',
    displayText: 'Bob Vance, Spouse',
    defaultPayeeCode: '10',
  },
  {
    value: '1129318238',
    displayText: 'Cathy Smith, Child',
    defaultPayeeCode: '11',
  },
];

describe('SelectClaimant', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const setVeteranIsNotClaimant = jest.fn();

  const defaultProps = {
    formType: 'appeal',
    relationships: defaultRelationships,
    featureToggles: { hlrScUnrecognizedClaimants: true },
    setVeteranIsNotClaimant
  };

  const setupDefault = (props = { ...defaultProps }) => {
    return render(<SelectClaimant {...props} />);
  };

  const renderSelectClaimant = (
    props = {
      toggled: false,
      formType: '',
    }
  ) => {
    return render(
      <SelectClaimant
        isVeteranDeceased
        relationships={defaultRelationships}
        benefitType={props.benefitType}
        formType={props.formType}
        veteranIsNotClaimant={props.veteranIsNotClaimant}
        setVeteranIsNotClaimant={setVeteranIsNotClaimant}
        featureToggles={props.featureToggles}
      />
    );
  };

  describe('with default value props', () => {
    it('renders correctly', () => {
      const { container } = setupDefault();

      expect(container).toMatchSnapshot();
    });

    it('passes a11y testing', async () => {
      const { container } = setupDefault();

      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('with formType', () => {
    const featureToggles = { hlrScUnrecognizedClaimants: true };
    const setupProps = { toggled: true, featureToggles };

    it('renders correctly', () => {
      const { container } = renderSelectClaimant(setupProps);

      expect(container).toMatchSnapshot();
    });

    it('passes a11y testing', async () => {
      const { container } = renderSelectClaimant(setupProps);

      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe("with formType 'appeal'", () => {
    const formType = { formType: 'appeal' };
    const featureToggles = { hlrScUnrecognizedClaimants: true };
    const setupProps = { ...formType, toggled: true, featureToggles };

    it('disables different-claimant-option_false radio button and does NOT fire off setVeteranIsNotClaimant', () => {
      renderSelectClaimant(setupProps);

      const radioNo = screen.getByRole('radio', { name: /no/i });

      expect(radioNo).toBeEnabled();
      expect(setVeteranIsNotClaimant).toBeCalledTimes(0);
    });

    it('renders deceasedVeteranAlert', () => {
      renderSelectClaimant({ ...setupProps, veteranIsNotClaimant: false });

      const alert = screen.getByRole('alert');

      expect(alert).toBeInTheDocument();
    });

    describe('nonVeteranClaimants enabled', () => {
      it('renders correctly', async () => {
        // Component only differs when veteranIsNotClaimant is set
        const veteranIsNotClaimant = true;
        const { container } = setupDefault({
          ...defaultProps,
          veteranIsNotClaimant,
        });

        // Ensure it's rendering the additional content
        expect(
          screen.queryByText(COPY.CLAIMANT_NOT_FOUND_START)
        ).not.toBeInTheDocument();
        expect(
          screen.queryByText(COPY.CLAIMANT_NOT_FOUND_END)
        ).not.toBeInTheDocument();

        expect(
          screen.queryByText(COPY.SELECT_NON_LISTED_CLAIMANT_LABEL)
        ).toBeInTheDocument();
        expect(
          screen.queryByRole('radio', { name: /claimant not listed/i })
        ).toBeInTheDocument();

        expect(container).toMatchSnapshot();
      });

      it('passes a11y testing', async () => {
        const { container } = setupDefault({ ...defaultProps });

        const results = await axe(container);

        expect(results).toHaveNoViolations();
      });
    });
  });

  describe("with formType not 'appeal'", () => {
    const featureToggles = { hlrScUnrecognizedClaimants: true };
    const setupProps = { toggled: true, featureToggles };

    it('disables different-claimant-option_false radio button & fires off setVeteranIsNotClaimant', () => {
      renderSelectClaimant(setupProps);

      const radioNo = screen.getByRole('radio', { name: /no/i });

      expect(radioNo).toBeDisabled();
      expect(setVeteranIsNotClaimant).toBeCalled();
    });
  });

  describe("with formType 'higher_level_review'", () => {
    const formType = 'higher_level_review';
    const veteranIsNotClaimant = true;

    it('short label, no "Claimant Not Listed" option and payee code dropdown for one of first 3 benefit types', () => {
      const benefitType = 'pension';
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType };

      renderSelectClaimant(setupProps);

      expect(screen.queryByText(COPY.SELECT_CLAIMANT_LABEL)).toBeInTheDocument();
      expect(screen.queryByText(COPY.SELECT_NON_LISTED_CLAIMANT_LABEL)).not.toBeInTheDocument();
      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).not.toBeInTheDocument();
      expect(screen.queryByText('What is the payee code for this claimant?')).toBeInTheDocument();
    });

    it('long label, "Claimant Not Listed" option, no payee code dropdown if other than first 3 benefit types', () => {
      const benefitType = 'insurance';
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType };

      renderSelectClaimant(setupProps);

      expect(screen.queryByText(COPY.SELECT_NON_LISTED_CLAIMANT_LABEL)).toBeInTheDocument();
      expect(screen.queryByText(COPY.SELECT_CLAIMANT_LABEL)).not.toBeInTheDocument();
      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).toBeInTheDocument();
      expect(screen.queryByText('What is the payee code for this claimant?')).not.toBeInTheDocument();
    });

    it('shows feature toggle is functioning correctly', () => {
      const featureToggles = { hlrScUnrecognizedClaimants: false };
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType: 'education', featureToggles };

      renderSelectClaimant(setupProps);

      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).not.toBeInTheDocument();
    });
  });

  describe("with formType 'supplemental_claim'", () => {
    const formType = 'supplemental_claim';
    const veteranIsNotClaimant = true;

    it('short label, no "Claimant Not Listed" option and payee code dropdown for one of first 3 benefit types', () => {
      const benefitType = 'fiduciary';
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType };

      renderSelectClaimant(setupProps);

      expect(screen.queryByText(COPY.SELECT_CLAIMANT_LABEL)).toBeInTheDocument();
      expect(screen.queryByText(COPY.SELECT_NON_LISTED_CLAIMANT_LABEL)).not.toBeInTheDocument();
      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).not.toBeInTheDocument();
      expect(screen.queryByText('What is the payee code for this claimant?')).toBeInTheDocument();
    });

    it('long label, "Claimant Not Listed" option, no payee code dropdown if other than first 3 benefit types', () => {
      const benefitType = 'loan_guaranty';
      const featureToggles = { hlrScUnrecognizedClaimants: true };
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType, featureToggles };

      renderSelectClaimant(setupProps);

      expect(screen.queryByText(COPY.SELECT_NON_LISTED_CLAIMANT_LABEL)).toBeInTheDocument();
      expect(screen.queryByText(COPY.SELECT_CLAIMANT_LABEL)).not.toBeInTheDocument();
      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).toBeInTheDocument();
      expect(screen.queryByText('What is the payee code for this claimant?')).not.toBeInTheDocument();
    });

    it('shows feature toggle is functioning correctly', () => {
      const benefitType = 'loan_guaranty';
      const featureToggles = { hlrScUnrecognizedClaimants: false };
      const setupProps = { ...defaultProps, formType, veteranIsNotClaimant, benefitType, featureToggles };

      renderSelectClaimant(setupProps);

      expect(screen.queryByRole('radio', { name: /claimant not listed/i })).not.toBeInTheDocument();
    });
  });
});
