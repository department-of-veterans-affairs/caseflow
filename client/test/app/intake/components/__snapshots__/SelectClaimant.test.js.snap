// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectClaimant with default value props renders correctly 1`] = `
<div>
  <div
    class="cf-different-claimant"
    style="margin-top: 10px;"
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Is the claimant someone other than the Veteran?
             
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="different-claimant-option_false"
            name="different-claimant-option"
            type="radio"
            value="false"
          />
          <label
            class=""
            for="different-claimant-option_false"
          >
            No
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="different-claimant-option_true"
            name="different-claimant-option"
            type="radio"
            value="true"
          />
          <label
            class=""
            for="different-claimant-option_true"
          >
            Yes
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`SelectClaimant with formType 'appeal' nonVeteranClaimants enabled renders correctly 1`] = `
<div>
  <div
    class="cf-different-claimant"
    style="margin-top: 10px;"
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Is the claimant someone other than the Veteran?
             
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="different-claimant-option_false"
            name="different-claimant-option"
            type="radio"
            value="false"
          />
          <label
            class=""
            for="different-claimant-option_false"
          >
            No
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            id="different-claimant-option_true"
            name="different-claimant-option"
            type="radio"
            value="true"
          />
          <label
            class=""
            for="different-claimant-option_true"
          >
            Yes
          </label>
        </div>
      </div>
    </fieldset>
    <div>
      <fieldset
        class="usa-fieldset-inputs cf-form-radio "
      >
        <legend
          class=""
        >
          <strong>
            <span>
              <p
                id="nonListedClaimantLabel"
                style="margin-top: 8.95px; margin-bottom: -25px;"
              >
                Please select the claimant listed on the form. If the claimant is not listed, please select "Claimant not listed" and add their information in the next step.
                <br />
                <br />
              </p>
               
            </span>
          </strong>
        </legend>
        <div
          class="cf-form-radio-options"
        >
          <div
            class="cf-form-radio-option"
          >
            <input
              id="claimant-options_CLAIMANT_WITH_PVA_AS_VSO"
              name="claimant-options"
              type="radio"
              value="CLAIMANT_WITH_PVA_AS_VSO"
            />
            <label
              class=""
              for="claimant-options_CLAIMANT_WITH_PVA_AS_VSO"
            >
              Bob Vance, Spouse
            </label>
          </div>
          <div
            class="cf-form-radio-option"
          >
            <input
              id="claimant-options_1129318238"
              name="claimant-options"
              type="radio"
              value="1129318238"
            />
            <label
              class=""
              for="claimant-options_1129318238"
            >
              Cathy Smith, Child
            </label>
          </div>
          <div
            class="cf-form-radio-option"
          >
            <input
              id="claimant-options_claimant_not_listed"
              name="claimant-options"
              type="radio"
              value="claimant_not_listed"
            />
            <label
              class=""
              for="claimant-options_claimant_not_listed"
            >
              Claimant not listed
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`SelectClaimant with formType renders correctly 1`] = `
<div>
  <div
    class="cf-different-claimant"
    style="margin-top: 10px;"
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Is the claimant someone other than the Veteran?
             
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            disabled=""
            id="different-claimant-option_false"
            name="different-claimant-option"
            type="radio"
            value="false"
          />
          <label
            class="disabled"
            for="different-claimant-option_false"
          >
            No
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="different-claimant-option_true"
            name="different-claimant-option"
            type="radio"
            value="true"
          />
          <label
            class=""
            for="different-claimant-option_true"
          >
            Yes
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
